tests/tf/policies/test_policies.py

```diff
@@ -19,6 +22,11 @@
 
 
 class TestTfPolicies(unittest.TestCase):
+    def setUp(self):
+        self.sess = tf.Session(graph=tf.Graph())
+        self.sess.__enter__()
```

Comment:
Please avoid using sess.__enter__()

Reply:
Why?

Reply:
There are nicer ways to handle tensorflow session lifecycle. `__enter__` and `__exit__` are mostly used for the `with` statement. Using `__enter__`, especially without further releasing the resources, is not a very clean way to use tensorflow sessions(even though we used it before). 

You can look up [with statement](https://docs.python.org/3/reference/compound_stmts.html#the-with-statement) and [tf.Session](https://github.com/tensorflow/tensorflow/blob/r1.10/tensorflow/python/client/session.py)

Reply:
ZP is correct in general but I think this is exactly actually what you want in a unit test. 

Rules are often bent in unit tests to make the tests easier to write and read.

A good way to deal with this is to add a custom class (inherit from TestCase) which overloads setUp/tearDown to always make a clean session and enter. I think TF already has one of these in the tf.tests, but unsure if it fits our use case.

Reply:
I think we should at least add an `__exit__` in tearDown.

Reply:
I've already tried tf.tests.TestCase.test_session() before but not work.
I agree we should add sess.close in tearDown.

Reply:
Done.